### DISCUSSION TITLE
Watchtower running on unraid but containers still say update after it runs

### DISCUSSION THREAD

**Discussion by lvunraider** [containrrr/watchtower#1389] [Q&A]
I'm not sure what I'm missing to get the containers to update automatically.
I get confirmation that they have been updated but when I check unraid they are still yellow and say "apply update"

Here are the post arguments I have:  --debug --rolling-restart --schedule "0 0 4 * * *" --cleanup --remove-volumes --include-restarting --notifications=email --include-stopped --revive-stopped.

Please let me know if you need any more info.

Thanks!

**Comment by simonaronsson** [comment]
Can you please post some logs of when this happens? You can find them using `docker logs watchtower`.

**Reply by Shurov** [reply]
Not the OP, but I have the same problem. Here are my exemplary logs:
```
time="2023-02-23T04:00:16-08:00" level=info msg="Found new esphome/esphome:latest image (3ceb484d7e0a)"
time="2023-02-23T04:00:34-08:00" level=info msg="Found new portainer/portainer-ce:latest image (a87d51c7a324)"
time="2023-02-23T04:00:47-08:00" level=info msg="Stopping /Portainer (4e1490ff03f7) with SIGTERM"
time="2023-02-23T04:00:49-08:00" level=info msg="Stopping /esphome (8fc5ad220b55) with SIGTERM"
time="2023-02-23T04:01:01-08:00" level=info msg="Creating /esphome"
time="2023-02-23T04:01:04-08:00" level=info msg="Creating /Portainer"
time="2023-02-23T04:01:08-08:00" level=info msg="Removing image d75324b2879e"
time="2023-02-23T04:01:09-08:00" level=info msg="Removing image 9b183f75f721"
time="2023-02-23T04:01:09-08:00" level=info msg="Session done" Failed=0 Scanned=16 Updated=2 notify=no
time="2023-02-24T04:00:26-08:00" level=info msg="Session done" Failed=0 Scanned=17 Updated=0 notify=no
```
So, on Feb 24th nothing was upgraded, however on Feb 23rd two containers were upgraded, while Unraid still shows "apply update" near them.

UPD: noticed that after server reboot containers' status updated accordingly, and no "apply update" is shown any more.
Hard refreshing the page (Ctrl+Shift+R, effectively clearing the cache) doesn't help, so only the server (physical machine) restart helps

**Comment by AN1MATEK** [comment]
I am also having this issue and posted in the UNRAID forums for help. Nobody tuned-in though.

https://forums.unraid.net/topic/142238-containers-show-update-available-even-when-it-is-up-to-date/

When you click on Update in UNRAID, it pulls 0 Bytes but restarts the container.

**Comment by thorrrr** [comment]
I have same issue showing loads of updates so i have to keep re ruining manually !
```
time="2023-07-23T13:31:44+01:00" level=info msg="Watchtower 1.5.3"
time="2023-07-23T13:31:44+01:00" level=info msg="Using no notifications"
time="2023-07-23T13:31:44+01:00" level=info msg="Checking all containers (except explicitly disabled with label)"
time="2023-07-23T13:31:44+01:00" level=info msg="Scheduling first run: 2023-07-24 04:00:00 +0100 BST"
time="2023-07-23T13:31:44+01:00" level=info msg="Note that the first check will be performed in 14 hours, 28 minutes, 15 seconds"
time="2023-07-24T04:00:24+01:00" level=info msg="Found new hotio/radarr:nightly image (84e7ceea2b2d)"
time="2023-07-24T04:00:47+01:00" level=info msg="Stopping /radarr (4d57290011a2) with SIGTERM"
time="2023-07-24T04:00:52+01:00" level=info msg="Creating /radarr"
time="2023-07-24T04:00:52+01:00" level=info msg="Removing image ea9482b31854"
time="2023-07-24T04:00:52+01:00" level=info msg="Session done" Failed=0 Scanned=28 Updated=1 notify=no
time="2023-07-25T04:00:24+01:00" level=info msg="Found new hotio/prowlarr:testing image (5bcb5485b319)"
time="2023-07-25T04:00:46+01:00" level=info msg="Stopping /prowlarr (594d9a3b17eb) with SIGTERM"
time="2023-07-25T04:00:51+01:00" level=info msg="Creating /prowlarr"
time="2023-07-25T04:00:52+01:00" level=info msg="Removing image 2a655f1b73cc"
time="2023-07-25T04:00:52+01:00" level=info msg="Session done" Failed=0 Scanned=28 Updated=1 notify=no
```

https://i.imgur.com/8Ed1T5t.png

**Comment by lephtHanded** [comment]
This isn't a watchtowrrr problem, it's updating the containers properly.
The issue is how unraid checks the containers/images.  

I use unraid myself but mainly use docker compose for my containers or stacks in portainer.
It's never shown that I'm running the latest version.

You can set portainer to show if you're running the latest image.

**Reply by piksel** [reply]
Yeah, this is what I suspect as well. The concept of "updating" a container doesn't really exist in docker (hence why watchtower was created in the first place). So Unraid probably doesn't check if a container has magically been updated and change it's UI.
